FirebaseSessions/Sources/SessionInitiator.swift

```diff
@@ -22,8 +28,72 @@ import Foundation
 ///      and comes to the foreground.
 ///
 class SessionInitiator {
+  let sessionTimeout: TimeInterval = 30 * 60 // 30 minutes
+  let currentTime: () -> Date
+  var backgroundTime = Date.distantFuture
+  var initiateSessionStart: () -> Void = {}
+
+  init(dateProvider: @escaping () -> Date = Date.init) {
```

Comment:
Slight nit - what do you think of calling this currentDateProvider to convey it should represent the current time?